frontends/common/resolveReferences/resolveReferences.h

```diff
@@ -57,37 +73,58 @@ class ResolutionContext : public IHasDbPrint {
     }
     void done();
 
-    // Resolve a reference for the specified name.
-    // The reference is restricted to be to an object of the specified type
-    // If previousOnly is true, the reference must precede the point of the 'name' in the program
+    /// Resolve references for @p name, restricted to @p type declarations.
+    /// If @p forwardOK is `false`, the referenced location must precede the location of @p name.
     std::vector<const IR::IDeclaration*>*
-    resolve(IR::ID name, ResolutionType type, bool previousOnly) const;
+    resolve(IR::ID name, ResolutionType type, bool forwardOK) const;
 
-    // Resolve a reference for the specified name; expect a single result
+    /// Resolve reference for @p name, restricted to @p type declarations, and expect one result.
+    /// If @p forwardOK is `false`, the referenced location must precede the location of @p name.
     const IR::IDeclaration*
-    resolveUnique(IR::ID name, ResolutionType type, bool previousOnly) const;
+    resolveUnique(IR::ID name, ResolutionType type, bool forwardOK) const;
 
-    // Resolve a Type_Name to a concrete type
+    // Resolve a refrence to a type @p type.
     const IR::Type *resolveType(const IR::Type *type) const;
 };
 
-// No prerequisites, but it usually must be run over the whole program.
-// Builds output in refMap.
-// The ReferenceMap maps each Path to a declaration.
+/** Inspector that computes `refMap`: a map from paths to declarations.
+ *
+ * @pre: None
+ *
+ * @post: produces an up-to-date `refMap`
+ *
+ * @todo: is @p rootNamespace redundant, since @p context always has it?
+ */
 class ResolveReferences : public Inspector {
-    // Output: reference map
+    /// Reference map -- essentially from paths to declarations.
     ReferenceMap* refMap;
+
+    /// Helper data structure that maintains current context.
     ResolutionContext* context;
+
+    /// The program's root namespace.
     const IR::INamespace* rootNamespace;
-    // used as a stack
-    std::vector<bool> resolveForward;  // if true allow resolution with declarations that follow use
+
+    /// Tracks whether forward references are permitted in a context.
+    std::vector<bool> resolveForward;
+
+    /// Indicates if _all_ forward references are allowed
     bool anyOrder;
+
+    /// Indicates if it should be an error for one declaration to shadow another.
```

Comment:
Actually it indicates whether warnings should be emitted for shadowing.
It is never an error.